src/psyclone/tests/domain/lfric/lfric_builtins_test.py

```diff
@@ -1608,6 +1608,154 @@ def test_inc_a_minus_X(tmpdir, monkeypatch, annexed, dist_mem, fortran_writer):
         assert output_dm in code
 
 
+def test_X_minus_a(tmpdir, monkeypatch, annexed, dist_mem, fortran_writer):
+    ''' Test that 1) the str method of LFRicAMinusXKern returns the
+    expected string and 2) we generate correct code for the built-in
+    operation Y = X - a where 'a' is a real scalar and X and Y
+    are real-valued fields. Test with and without annexed dofs being
+    computed as this affects the generated code.
+    Also test the lower_to_language_level() method.
+
+    '''
+    api_config = Config.get().api_conf(API)
+    monkeypatch.setattr(api_config, "_compute_annexed_dofs", annexed)
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "15.2.9_X_minus_a_builtin.f90"),
+                           api=API)
+
+    psy = PSyFactory(API, distributed_memory=dist_mem).create(invoke_info)
+    # Test string method
+    first_invoke = psy.invokes.invoke_list[0]
+    kern = first_invoke.schedule.children[0].loop_body[0]
+    assert str(kern) == "Built-in: X_minus_a (real-valued fields)"
+    # Test code generation
+    code = str(psy.gen)
+
+    assert LFRicBuild(tmpdir).code_compiles(psy)
+
+    # Check for the correct 'use' module statements
+    output_mod = (
+        "    USE constants_mod, ONLY: r_tran, i_def\n"
+        "    USE r_tran_field_mod, ONLY: r_tran_field_type, "
+        "r_tran_field_proxy_type\n")
+    assert output_mod in code
+
+    if not dist_mem:
+        output = (
+            "      loop0_stop = undf_aspc1_f2\n"
+            "      !\n"
+            "      ! Call our kernels\n"
+            "      !\n"
+            "      DO df=loop0_start,loop0_stop\n"
+            "        f2_proxy%data(df) = f1_proxy%data(df) - a\n"
+            "      END DO\n"
+            "      !\n"
+            "    END SUBROUTINE invoke_0\n")
+        assert output in code
+
+        # Test the lower_to_language_level() method
+        kern.lower_to_language_level()
+        loop = first_invoke.schedule.walk(Loop)[0]
+        code = fortran_writer(loop)
+        assert ("do df = loop0_start, loop0_stop, 1\n"
+                "  f2_proxy%data(df) = f1_proxy%data(df) - a\n"
+                "enddo") in code
+    else:
+        output_dm = (
+            "      loop0_start = 1\n"
+            "      loop0_stop = f2_proxy%vspace%get_last_dof_annexed()\n"
```

Comment:
I think I must be missing something but how come this works, irrespective of the value of `annexed`? When `annexed` is `False` I would expect to see `get_last_dof_owned()` here.

Reply:
It works because of lines 1679 and 1680 below:
```
        if not annexed:
            output_dm = output_dm.replace("dof_annexed", "dof_owned")
```

Reply:
Oh yes! Doh!